lib/enum_type.ex

```diff
@@ -113,7 +116,7 @@ defmodule EnumType do
 
       def value(unquote(option)), do: unquote(option).value
 
-      with {:module, _module} <- Code.ensure_compiled(Ecto.Type) do
+      if unquote(generate_ecto_type?(true)) do
```

Comment:
Ideally we wouldn't generate these functions as well when `generate_ecto_type` is passed, but I'm not sure how viable it is to pass `generate_ecto_type` from `defenum` to here (and at least the functions can be ignored).